src/safeposix/syscalls/fs_calls.rs

```diff
@@ -441,25 +475,39 @@ impl Cage {
                 let newinodenum = FS_METADATA
                     .nextinode
                     .fetch_add(1, interface::RustAtomicOrdering::Relaxed); //fetch_add returns the previous value, which is the inode number we want
-                if let Inode::Dir(ref mut parentdir) =
-                    *(FS_METADATA.inodetable.get_mut(&pardirinode).unwrap())
+                
+                // Insert a reference to the file in the parent directory and update the inode attributes
+                // Fetch the inode of the parent directory and only proceed when its type is directory.
+                if let Inode::Dir(ref mut parentdir) = *(FS_METADATA.inodetable.get_mut(&pardirinode).unwrap())
```

Comment:
Are you undoing `cargo fmt`?

Reply:
Sorry, I missed this change. Have reverted it. 
I found that a few other formatting changes in a couple of files, including this one, need to be updated according to the "Rust format". We can probably have a small bug for it to get them fixed together.